api/src/org/labkey/api/data/NameGenerator.java

```diff
@@ -224,73 +657,173 @@ private void initialize(@Nullable Map<String, String> importAliases)
+
+                                lookupValuePreview = getLineageLookupTokenPreview(fkTok, fieldParts.get(0), null, fieldParts.get(1), user);
+                            }
+                            else if (fieldParts.size() == 3)
+                            {
+                                // MaterialInputs/SampleType/lookup, DataInputs/DataClass/lookup
+                                lineageLookupFields.computeIfAbsent(fieldParts.get(0) + "/" + fieldParts.get(1), (s) -> new ArrayList<>()).add(fieldParts.get(2));
+                                lookupValuePreview = getLineageLookupTokenPreview(fkTok, fieldParts.get(0), fieldParts.get(1), fieldParts.get(2), user);
+                            }
                         }
-                        else if (fieldParts.size() == 3)
+                        else
                         {
-                            // MaterialInputs/SampleType/lookup, DataInputs/DataClass/lookup
-                            lineageLookupFields.computeIfAbsent(fieldParts.get(0) + "/" + fieldParts.get(1), (s) -> new ArrayList<>()).add(fieldParts.get(2));
+                            String errorMsg = "Only one level of lookup supported for lineage input: " + fkTok + ".";
+                            if (_validateSyntax)
+                                _syntaxErrors.add(errorMsg);
+                            else
+                                throw new UnsupportedOperationException(errorMsg);
                         }
-                        else
-                            throw new UnsupportedOperationException("Only one level of lookup supported for lingeage input: " + fkTok);
+
+                        if (lookupValuePreview != null)
+                            previewCtx.put(fkTok.toString(), lookupValuePreview);
+
+                        continue;
                     }
                     else if (fieldParts.size() > 2)
                     {
                         // for now, we only support one level of lookup: ${ingredient/name}
                         // future versions could support multiple levels
-                        throw new UnsupportedOperationException("Only one level of lookup supported: " + fkTok);
+                        String errorMsg = "Only one level of lookup supported: " + fkTok + ".";
+                        if (_validateSyntax)
+                            _syntaxErrors.add(errorMsg);
+                        else
+                            throw new UnsupportedOperationException(errorMsg);
                     }
 
-                    if (_parentTable == null)
-                        throw new UnsupportedOperationException("Parent table required for name expressions with lookups: " + fkTok);
+                    if (_parentTable == null && domainFields.isEmpty())
+                    {
+                        String errorMsg = "Parent table required for name expressions with lookups: " + fkTok + ".";
```

Comment:
```suggestion
                        String errorMsg = "Parent table is required for name expressions with lookups: " + fkTok + ".";
```
I worry a little about these error messages referring to "Parent table" and "lookup" when that isn't wording we use in LKSM.  And we're using "name expressions" here instead of "naming patterns".  But I don't have a better suggestion given inconsistencies in the products.